up-transport-vsomeip/src/listener_registry.rs

```diff
@@ -0,0 +1,314 @@
+/********************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use crate::message_conversions::convert_vsomeip_msg_to_umsg;
+use crate::{ApplicationName, AuthorityName, ClientId};
+use cxx::{let_cxx_string, SharedPtr};
+use lazy_static::lazy_static;
+use log::{error, info, trace};
+use once_cell::sync::Lazy;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::sync::{mpsc, Arc};
+use tokio::runtime::Runtime;
+use tokio::sync::RwLock;
+use tokio::task::LocalSet;
+use tokio::time::Instant;
+use up_rust::{ComparableListener, UListener, UUri};
+use up_rust::{UCode, UMessage, UStatus};
+use vsomeip_proc_macro::generate_message_handler_extern_c_fns;
+use vsomeip_sys::glue::{make_application_wrapper, make_message_wrapper, make_runtime_wrapper};
+use vsomeip_sys::safe_glue::get_pinned_runtime;
+use vsomeip_sys::vsomeip;
+
+const THREAD_NUM: usize = 10;
+
+// Create a separate tokio Runtime for running the callback
+lazy_static! {
+    static ref CB_RUNTIME: Runtime = tokio::runtime::Builder::new_multi_thread()
+        .worker_threads(THREAD_NUM)
+        .enable_all()
+        .build()
+        .expect("Unable to create callback runtime");
+}
+
+static RUNTIME: Lazy<Arc<Runtime>> =
+    Lazy::new(|| Arc::new(Runtime::new().expect("Failed to create Tokio runtime")));
+
+fn get_runtime() -> Arc<Runtime> {
+    Arc::clone(&RUNTIME)
+}
+
+generate_message_handler_extern_c_fns!(10000);
+
+type ListenerIdMap = RwLock<HashMap<(UUri, Option<UUri>, ComparableListener), usize>>;
+lazy_static! {
+    static ref LISTENER_ID_CLIENT_ID_MAPPING: RwLock<HashMap<usize, ClientId>> =
+        RwLock::new(HashMap::new());
```

Comment:
IMHO all of these could simply be (less expensive) std RwLocks, or am I mistaken?

Reply:
Yeah, changed to use std::sync::RwLock

FWIW after changing, didn't notice any performance change in terms of throughput of messages.